Plugins/Public/base_plugin/PlayerBase.cpp

```diff
@@ -677,7 +677,16 @@ float PlayerBase::SpaceObjDamaged(uint space_obj, uint attacking_space_obj, floa
 			{
 				if (set_plugin_debug > 1)
 					ConPrint(L"PlayerBase::damaged space_obj=%u\n", space_obj);
-				hostile_tags[charname] = charname;
+				
+				if (hostile_tags_damage.find(charname) != hostile_tags_damage.end())
+				{
+					if (hostile_tags_damage[charname] < damage_treshold)
+						hostile_tags_damage[charname] += damage;
+					else hostile_tags[charname] = charname;
+				}
+				else hostile_tags_damage[charname] = damage;
```

Comment:
What if damage >= damage_threshold in this case?

Reply:
lets see....
loop 1: not found data, setting hostile_tags_damage[charname] = damage;
loop 2: data has been found, hostile_tags_damage[charname] is not less than damage threshold, which means hostile_tags[charname] = charname; (declare target as enemy)

all is fine. if someone made more damage than damage threshold, he will be declared as enemy during his second shot.